packages/react-native-windows-init/src/Cli.ts

```diff
@@ -274,36 +277,53 @@ async function getLatestMatchingRNWVersion(
   }
 }
 
-function installReactNativeWindows(version: string, useDevMode: boolean) {
-  console.log(
-    `Installing ${chalk.green('react-native-windows')}@${chalk.cyan(
-      version,
-    )}...`,
-  );
-
+function installReactNativeWindows(
+  version: string | undefined,
+  useDevMode: boolean,
+) {
   const cwd = process.cwd();
   const execOptions = argv.verbose ? {stdio: 'inherit' as 'inherit'} : {};
 
   if (useDevMode) {
-    execSync(
-      isProjectUsingYarn(cwd)
-        ? 'yarn link react-native-windows'
-        : 'npm link react-native-windows',
-      execOptions,
+    const packageCmd = isProjectUsingYarn(cwd) ? 'yarn' : 'npm';
+    execSync(`${packageCmd} link react-native-windows`, execOptions);
+    const rnwPkgJsonPath = path.resolve(
+      './node_modules/react-native-windows/package.json',
     );
-  } else {
-    const pkgJsonPath = findUp.sync('package.json', {cwd});
-    if (!pkgJsonPath) {
-      throw new Error('Unable to find package.json');
+    const rnwVersion = require(rnwPkgJsonPath).version;
+    if (version && version !== rnwVersion) {
+      console.error(
+        chalk.redBright(
+          `[Error] Requested react-native-windows version: '${version}' does not match version '${rnwVersion}' of the linked module. When using '--useDevMode' you do not need to pass a version. If you do, you should pass '--version ${rnwVersion}'`,
+        ),
+      );
+      process.exit(EXITCODE_DEVMODE_VERSION_MISMATCH);
+    } else if (!version) {
+      version = rnwVersion;
     }
+  } else if (!version) {
+    throw new Error(
+      'Unexpected error. Version is expected to be set when not using devMode.',
```

Comment:
Nit: This error message might be confusing if a user sees it who isn't familiar with devMode. Consider a more generic error message, You can also add validation steps to yargs where you could check this.